opentracing-impl/src/test/java/io/opentracing/AbstractTracerTest.java

```diff
@@ -89,6 +89,23 @@ public void testNonEmptyExtract() {
         assertEquals("Should find the marker", "whatever", ((TestSpanBuilder)result).operationName);
     }
 
+    @Test
+    public void testExtractAsParent() throws Exception {
+        Map<String,String> map = Collections.singletonMap("test-marker", "whatever");
+        TextMapExtractAdapter adapter = new TextMapExtractAdapter(map);
+        AbstractTracer tracer = new TestTracerImpl();
+        SpanContext parent = tracer.extract(Format.Builtin.TEXT_MAP, adapter);
+        assert AbstractNoopSpan.INSTANCE != tracer.buildSpan("child").asChildOf(parent).start();
+    }
+
+    @Test
+    public void testExtractOfNoParent() throws Exception {
+        AbstractTracer tracer = new TestTracerImpl();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf((Span)AbstractNoopSpan.INSTANCE).start();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf((SpanContext)AbstractNoopSpan.INSTANCE).start();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf(AbstractNoopSpanBuilder.INSTANCE).start();
```

Comment:
I am not sure I follow this. The way TestTextMapExtractorImpl is written it returns a NOOP span context if the carrier does not contain a "trace context" (conceptually). Then here we validate that by using the noop span context as a parent we get back a noop span (it's also the title of the PR). But if we ignore the noop part of it, what it boils down is if my server receives a request without tracing info, it does not start a new trace. I think the most common behavior of the tracers is actually the opposite: continue the trace if there is one, start a new one otherwise.